src/main/java/by/exadel/internship/controller/AuthController.java

```diff
@@ -0,0 +1,57 @@
+package by.exadel.internship.controller;
+
+import by.exadel.internship.config.jwt.JwtUtils;
+import by.exadel.internship.entity.UserDetailsImpl;
+import by.exadel.internship.pojo.JwtResponse;
+import by.exadel.internship.pojo.LoginRequest;
+import io.swagger.annotations.Api;
+import io.swagger.annotations.ApiOperation;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.ResponseEntity;
+import org.springframework.security.authentication.AuthenticationManager;
+import org.springframework.security.authentication.UsernamePasswordAuthenticationToken;
+import org.springframework.security.core.Authentication;
+import org.springframework.security.core.context.SecurityContextHolder;
+import org.springframework.web.bind.annotation.*;
+
+import java.util.List;
+import java.util.stream.Collectors;
+
+@RestController
+@RequestMapping("/auth")
+@Api(tags = "Authorization controller")
+public class AuthController {
+    @Autowired
+    private AuthenticationManager authenticationManager;
+
+    @Autowired
+    JwtUtils jwtUtils;
+
+    @PostMapping("/signIn")
+    @ApiOperation("Authorize method")
+    public ResponseEntity<?> authUser(@RequestBody LoginRequest loginRequest){
+        splitEmail(loginRequest);
+        Authentication authentication = authenticationManager
+                .authenticate(new UsernamePasswordAuthenticationToken(
+                        loginRequest.getLogin(),
+                        loginRequest.getPassword()));
+        SecurityContextHolder.getContext().setAuthentication(authentication);
+        String jwt = jwtUtils.generateJwtToken(authentication);
+        UserDetailsImpl userDetails = (UserDetailsImpl) authentication.getPrincipal();
+        List<String> role = userDetails.getAuthorities().stream()
+                .map(item -> item.getAuthority())
+                .collect(Collectors.toList());
+        return ResponseEntity.ok(new JwtResponse(jwt,userDetails.getId(),
+                userDetails.getUsername(),
+                userDetails.getEmail(),
+                role));
+    }
+
+    private void splitEmail(LoginRequest loginRequest){
+        String[] emailLikeArray;
+        if(loginRequest.getLogin().contains("@")){
```

Comment:
For String manipulations use StringUtils

Reply:
Done